package.json

```diff
@@ -75,7 +75,7 @@
     "yargs": "^8.0.1"
   },
   "peerDependencies": {
-    "jest": "^20.0.0",
+    "jest": "^20.0.0 || ^21.0.0-beta.1",
```

Comment:
Based on the the issue, it seems like we also need to allow `^20.1.0-beta.1`. I think 

`^20.0.0 || ^20.1.0-alpha.1 || ^21.0.0-alpha.1`

is what we're looking for. This will allow the current major, next minor pre-release version and the next major pre-release version to be used.

Reply:
Do you mean `^20.0.0 || ^21.0.0-alpha.1`? I think `^20.0.0 || ^20.1.0-alpha.1` is equivalent to `^20.0.0 || ^21.0.0-beta.1` as far as semver is concerned, but I'm happy to make that change for clarity's sake.

Reply:
>I think ^20.0.0 || ^20.1.0-alpha.1 is equivalent to ^20.0.0 || ^21.0.0-beta.1

- The major/minor versions are different for the second terms. We can't use the next major version (prerelease or otherwise) because it will likely break things
- Using `alpha` will allow all prelease versions starting from `alpha`. Using `beta` will disallow the `alpha` versions.

This is my understanding from playing a bit with the semver calculator

Reply:
My understanding from playing with the semver calculator is that if you say `^21.0.0-foo` you are including any and all prerelease tags that could ever be appended to `21.0.0`, as well as any versions that would match `^21.0.0`.

Reply:
> My understanding from playing with the semver calculator is that if you say ^21.0.0-foo you are including any and all prerelease tags that could ever be appended to 21.0.0, as well as any versions that would match ^21.0.0

This sounds right. 

I think we might have different perspectives on what we need here. The way I look at it, we need to allow

- all the matching major/minor/patch versions. So in ts-jest 20, we want to allow jest versions `20.x.x`. `^20.0.0` takes care of this
- all the prerelease versions of next minor version. So if the current release of jest is `20.0.4`, we want to allow prerelease versions in the `20.1.0-foo` range. `^20.1.0-alpha.1` takes care of this

If we allow `^21.0.0-foo`, when the current release is `20.x.x`. it will most likely break because ts-jest maintains compatibility only with the corresponding jest versions. So ts-jest `20.x.x` is unlikely to be compatible with jest `21.x.x`. That's why there's no point in allowing `21.x.x` - prerelease or otherwise.

Reply:
> If we allow ^21.0.0-foo, when the current release is 20.x.x. it will most likely break because ts-jest maintains compatibility only with the corresponding jest versions. So ts-jest 20.x.x is unlikely to be compatible with jest 21.x.x. That's why there's no point in allowing 21.x.x - prerelease or otherwise.

Hm, the point of this PR from my perspective is to get rid of the warning that happens when I try to use `ts-jest` with `jest@test` (#282). Currently, `jest@test` = `jest@21.0.0.beta-1`. I am already using `ts-jest` extensively with `jest@21.0.0.beta-1` and it is fully compatible as far as I can tell, so I would like to open up the version range so that I can use them together without the warning.

Reply:
I see. Here's what I suggest

- setting the range to `^20.0.0 || ^20.1.0-alpha.1 || ^21.0.0-alpha.1`. This will cater to your use case and to the one mentioned on #282 (that specific version)
- adding a note to the README that makes it clear that using the next major version of jest might break things.

Does that sound reasonable?

Reply:
Just to clarify, adding `^20.1.0-alpha.1` will allow users to use the prerelease version within the current range if the one in the next major range breaks something for them

Reply:
Sure, I can do that.